pkg/authorization/authorizer/authorizer.go

```diff
@@ -52,40 +40,41 @@ func (a *openshiftAuthorizer) Authorize(attributes authorizer.Attributes) (bool,
 	return false, denyReason, nil
 }
 
+//TODO should we port the code here to another place. In order to obtain users/groups
 // GetAllowedSubjects returns the subjects it knows can perform the action.
 // If we got an error, then the list of subjects may not be complete, but it does not contain any incorrect names.
 // This is done because policy rules are purely additive and policy determinations
 // can be made on the basis of those rules that are found.
-func (a *openshiftSubjectLocator) GetAllowedSubjects(attributes authorizer.Attributes) (sets.String, sets.String, error) {
-	users := sets.String{}
-	groups := sets.String{}
-	namespace := attributes.GetNamespace()
-	subjects, err := a.delegate.AllowedSubjects(attributes)
-	for _, subject := range subjects {
-		switch subject.Kind {
-		case rbac.UserKind:
-			users.Insert(subject.Name)
-		case rbac.GroupKind:
-			groups.Insert(subject.Name)
-		case rbac.ServiceAccountKind:
-			// default the namespace to namespace we're working in if
-			// it's available. This allows rolebindings that reference
-			// SAs in the local namespace to avoid having to qualify
-			// them.
-			ns := namespace
-			if len(subject.Namespace) > 0 {
-				ns = subject.Namespace
-			}
-			if len(ns) >= 0 {
-				name := serviceaccount.MakeUsername(ns, subject.Name)
-				users.Insert(name)
-			}
-		default:
-			continue // TODO, should this add errs?
-		}
-	}
-	return users, groups, err
-}
+//func (a *openshiftSubjectLocator) GetAllowedSubjects(attributes authorizer.Attributes) (sets.String, sets.String, error) {
+//	users := sets.String{}
+//	groups := sets.String{}
+//	namespace := attributes.GetNamespace()
+//	subjects, err := a.delegate.AllowedSubjects(attributes)
```

Comment:
@abstractj sync with @tiran to use this as the foundation for two public functions:

1. `[]rbac.Subject -> users, groups []string`
2. `users, groups []string -> []rbac.Subject`

Put those in a new file at `pkg/authorization/registry/util/subject.go` for now.

Reply:
`[]rbac.Subject, defaultServiceAccountNamespace` -> `users, groups []string`

do we need the reverse?

Reply:
https://github.com/openshift/origin/pull/16033/files#diff-9cbb0b75c1c78e77c6ee106a6ac6fdd8R121